src/generic_ui/polymer/cloud-install.ts

```diff
@@ -130,5 +141,9 @@ Polymer({
   },
   ready: function() {
     this.ui = ui;
+    // ID of the latest attempt to create a server, used to distinguish
+    // between install failures that should be flagged to the user and
+    // failures owing to cancellation.
+    this.mostRecentCreateId = 0;
   }
 });
```

Comment:
Rather than relying on the `throw` site to throw a magic value, which we were checking for here to determine whether to show the failure overlay, we now track this state through an explicit variable directly within our control. This code is more robust, nice.

But I'm a bit unclear on the `createId` logic. What's the randint doing that a boolean can't? i.e. Rather than storing the createId, why can't we just initialize `this.userCanceled = false`, set it to `true` where we're currently setting the createId to 0, and then just check if it's `true` in the `catch()` callback?